examples/pruning/lightgbm_with_custom_cv_function.py

```diff
@@ -0,0 +1,404 @@
+        elif objective == "regression":
+            lgbm_cv_fun = LGBMRegressor
+
+        model = lgbm_cv_fun(n_estimators=n_estimators, random_state=random_state)
+
+        # pass hyperparameters dict to the estimator
+        model.set_params(**lgbm_params)
+
+        # train the model
+        model.fit(
+            X_train,
+            y_train.values.ravel(),
+            eval_set=(X_valid, y_valid.values.ravel()),
+            eval_metric=[eval_metric],  # note: a list required
+            eval_names=[eval_name],  # note: a list required
+            early_stopping_rounds=early_stopping_rounds,
+            verbose=verbose,
+            feature_name=feature_names,
+            callbacks=callbacks,
+        )
+
+        # collect current fold data
+        fold_best_iterations.append(model.best_iteration_)
+        fold_best_scores.append(model.best_score_[eval_name])
+
+    # average folds iterations numbers
+    folds_data = {}
+    folds_data["best_iterations_mean"] = -1
+    if fold_best_iterations[0] is not None:
+        folds_data["best_iterations_mean"] = int(np.mean(fold_best_iterations))
+
+    # collect metrics for best scores in each fold
+    fold_best_score = {}
+    for metric in fold_best_scores[0].keys():
+        fold_best_score[metric] = [fold[metric] for fold in fold_best_scores]
+
+    # avearage folds metrics (for all metrics)
+    for metric in fold_best_scores[0].keys():
+        folds_data["eval_mean-" + metric] = np.mean(fold_best_score[metric])
+
+    return {
+        "folds_mean_data": folds_data,
+        "feature_names": feature_names,
+        "fold_best_iter": fold_best_iterations,
+        "fold_best_score": fold_best_score,
+    }
+
+
+def load_sklearn_toy_data(
+    objective="binary", test_size=0.25, as_df=False, seed=123, verbose=False
+):
+
+    if objective in ["binary", "classification"]:
+        # load toy classification dataset (breast cancer wisconsin)
+        data, target = datasets.load_breast_cancer(return_X_y=True)
+
+    elif objective == "regression":
+        # load toy regression dataset (Boston house prices)
+        data, target = datasets.load_boston(return_X_y=True)
+
+    train_x, valid_x, train_y, valid_y = train_test_split(
+        data, target, test_size=test_size, shuffle=True, random_state=seed
+    )
+
+    if verbose:
+        print([i.shape for i in [train_x, valid_x, train_y, valid_y]])
+
+    if as_df:
+        # convert numpy arrays to DataFrames
+        train_x_df = pd.DataFrame(train_x)
+        train_x_df.columns = ["col_" + str(i) for i in train_x_df.columns]
+
+        valid_x_df = pd.DataFrame(train_x)
+        valid_x_df.columns = ["col_" + str(i) for i in valid_x_df.columns]
+
+        train_y_df = pd.DataFrame({"y": train_y})
+
+        valid_y_df = pd.DataFrame({"y": valid_y})
+
+        return train_x_df, valid_x_df, train_y_df, valid_y_df
+
+    else:
+        return train_x, valid_x, train_y, valid_y
+
+
+class Objective_custom:
```

Comment:
Please use camel case (c.f., [coding style convention](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#example))
```suggestion
class ObjectiveCustom:
```

Reply:
For consistency this change is also required in [#L349](https://github.com/mirekphd/optuna/blob/lgbm-w-cust-cv-fun/examples/pruning/lightgbm_with_custom_cv_function.py#L349)